tests/FSharp.Compiler.Service.Tests/FSharp.Compiler.Service.SurfaceArea.netstandard20.release.bsl

```diff
@@ -10179,6 +10179,7 @@ FSharp.Compiler.Text.ISourceText: Int32 GetLineCount()
 FSharp.Compiler.Text.ISourceText: Int32 Length
 FSharp.Compiler.Text.ISourceText: Int32 get_Length()
 FSharp.Compiler.Text.ISourceText: System.String GetLineString(Int32)
+FSharp.Compiler.Text.ISourceText: System.String GetSubTextFromRange(FSharp.Compiler.Text.Range)
```

Comment:
More thinking out loud here - we can make a return type to be a ReadOnlySpan<char> so it's compatible with any improvements we make in future (if we start spanifying the FCS). Currently it will still use the expensive strings manipulation. In future, we can just do spans.

Reply:
> It's worth noting that no APIs in FCS currently utilize Spans.

Yeah, but we should start doing it. There are literlly zero downsides of doing that.

> I prefer to handle all Span-related changes uniformly if and when they are integrated into FCS.

That would mean that we will have two APIs, one which returns Span and one which returns String. And we will have to obsolete one.

Reply:
**Update:** not saying we should do it now, but rather "we should start doing it".

Reply:
> Yeah, but we should start doing it. There are literlly zero downsides of doing that.

Meh, I feel like this misses a specific ask first. My use-case is to have this as a string.
As long as nobody else asks for a `Span` we shouldn't settle on `Span`. If in two months, someone has a use-case, by all means, break the API and change it to `Span`.

Reply:
> > Yeah, but we should start doing it. There are literlly zero downsides of doing that.
> 
> Meh, I feel like this misses a specific ask first. My use-case is to have this as a string. As long as nobody else asks for a `Span` we shouldn't settle on `Span`. If in two months, someone has a use-case, by all means, break the API and change it to `Span`.

The ask is simple really - in tooling scenarios, we need to allocate as little as possible. We have a use case for it in LSP.

Reply:
Sight, I'll play ball...

Reply:
> Sight, I'll play ball...

It can be done later, just saying that if we can use them, we probably should.